src/unittest/SymbolManager_test.cc

```diff
@@ -107,7 +115,7 @@ TEST_CASE("SymbolManager: detectType")
 TEST_CASE("SymbolManager: loadLines")
 {
 	auto dummyParser = [](std::span<std::string_view> tokens) -> std::optional<Symbol> {
-		if (tokens.size() == 1) return Symbol{std::string(tokens[0]), 123};
+		if (tokens.size() == 1) return Symbol{std::string(tokens[0]), 123, 0, 0};
 		return {};
 	};
 
```

Comment:
Thanks for expanding the unit-test.
Could you also add a test for the heuristic:
* if all segments in a file are 0, then assume the file has no information on segments.

In other words:
* Add a test where all segments are 0. This should result in no segments (segment is `std::nullopt`).
* Add a test with a mix of 0 and non-0 segments. Now the 0-segments should be 0 (not `std::nullopt`).